app/controllers/users_controller.rb

```diff
@@ -1,24 +1,29 @@
 class UsersController < ApplicationController
+  skip_before_action :check_interrupt_redirects
+
   def accept_transition_info
-    UpdateUserService.call(user, "accept_transition_screen!")
-    redirect_to Settings.rollover ? rollover_path : providers_path
+    user.accept_transition_screen!
```

Comment:
Was there any particular reason why the UpdateUserService is no longer being used?

Reply:
I might need @asmega to explain it well, but we found that using these `user.accept_transition_screen!` methods on the user was already updating the database object with the new state, thanks to the `!` on the end. It made the service call redundant. 

Basically the way the state machine works in the user model means changing the state does everything already

Reply:
Yes, I don't see why it was ever created it the first place or the purpose of it's existence. The class literally takes an object and method and sends that method to the object. Why don't we just invoke that method on the original object? It would literally take one line. Why do we need an entirely new class to invoke a method? For me it's smoke and mirrors and un-needed complexity for the sake of complexity. I wish to remove the `UpdateUserService` class as soon as possible.

Reply:
The UpdateUserService was extracted out as it the logic is required in more than one place. The service is doing more than simply calling the API, and has error handling. Misha and I have discussed and agreed it would be a good idea to ultimately remove the UpdateUserService and add the error handling to the Base class so that all api calls can benefit from it. I'll add a ticket to the ideas board, in the meantime might as well keep using it? Note that the NotificationsController is also calling this class

Reply:
https://github.com/DFE-Digital/publish-teacher-training/pull/1178/files#r442766309

Reply:
<img width="1556" alt="discussion" src="https://user-images.githubusercontent.com/5256922/87546516-97616c00-c6a1-11ea-86b0-b08f1e518d8a.png">

Reply:
In which case yes this belongs in the client otherwise you'll have to implement this on every API call

Reply:
But, why was this added in the first place? There must have been a specific error or case that is trying to be handled?

Reply:
Yeah not entirely sure regarding the historical context. I can discuss with Misha